### DISCUSSION TITLE
Dependencies

### DISCUSSION THREAD

**Discussion by CommonClimate** [LinkedEarth/Pyleoclim_util#450] [General]
Periodically we need to discuss issues surrounding dependencies, so this is the space to do it.

Latest installment:  a recent failed CI test on github (but success on my machine) and some sleuthing by @alexkjames  uncovered this uncomfortable truth:  our environment got befouled by a conflict between the latest SciPy (>=1.11.0) and Cartopy. The conflict is [known](https://github.com/SciTools/cartopy/issues/2199), but still awaiting resolution, and I don’t see any of us writing that fix.
Until it is done, my latest PR (#449, which now passes) has taken the drastic measure of forcing scipy=1.10.1. Needless to say, pinned versions are something we strive to avoid, but they are sometimes unavoidable, as in the present case. I encourage all of you to monitor the Cartopy issue and report back when you see a fix, so we can relax this pin. Thank you!

**Comment by fzhu2e** [comment]
After some tests, I realize that `pip` will try to install the latest version of the dependencies if not installed already, while `conda` will have some dependency check and fix processes. In our case, `scipy` is a dependency of `kneed` and `statsmodels`. If `pip install pyleoclim`, the latest version of `scipy` will be installed, causing the mentioned conflict. If we `conda install statsmodels` first (need to avoid `-c conda-forge`), `scipy=1.10.1` will be installed, avoiding the mentioned issue.

Therefore, a quick fix on this issue is install `statsmodels` via conda first before `pip install pyleoclim`. Note that this is already done in CI.
